src/frontend/packages/lib_components/src/utils/retryQuery.spec.ts

```diff
@@ -10,4 +10,9 @@ describe('retryQuery', () => {
     const result = retryQuery(1, { status: 403 });
     expect(result).toBe(false);
   });
+
+  it('should not retry on 500 errors after 3 failures', () => {
```

Comment:
You should probably simulate a loop, you expect than `retryQuery` has been called 3 times and at the end result is false